projects/igniteui-angular/src/lib/date-range-picker/igx-date-range-picker.component.ts

```diff
@@ -0,0 +1,696 @@
+ * ```
+ */
+@Component({
+    selector: 'igx-date-range-picker',
+    templateUrl: './igx-date-range-picker.component.html',
+    providers: [{ provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => IgxDateRangePickerComponent), multi: true }]
+})
+export class IgxDateRangePickerComponent extends DisplayDensityBase
+    implements IToggleView, OnChanges, AfterViewInit, OnDestroy, ControlValueAccessor {
+    /**
+     * Display calendar in either `dialog` or `dropdown` mode.
+     * @remarks
+     * Default mode is `dialog`
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker [mode]="dropdown"></igx-date-range-picker
+     * ```
+     */
+    @Input()
+    public mode = InteractionMode.Dialog;
+
+    /**
+     * The number of displayed month views.
+     *
+     * @remarks
+     * Default is `2`.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker [monthsViewNumber]="3"></igx-date-range-picker>
+     * ```
+     */
+    @Input()
+    public monthsViewNumber = 2;
+
+    /**
+     * Gets/Sets whether dates that are not part of the current month will be displayed.
+     *
+     * @remarks
+     * Default value is `false`.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker [hideOutsideDays]="true"></igx-date-range-picker>
+     * ```
+     */
+    @Input()
+    public hideOutsideDays: boolean;
+
+    /**
+     * The start day of the week.
+     *
+     * @remarks
+     * Can be assigned to a numeric value or to `WEEKDAYS` enum value.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker [weekStart]="1"></igx-date-range-picker>
+     * ```
+     */
+    @Input()
+    public weekStart = WEEKDAYS.SUNDAY;
+
+    /**
+     * The `locale` of the calendar.
+     *
+     * @remarks
+     * Default value is `"en"`.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker locale="jp"></igx-date-range-picker>
+     * ```
+     */
+    @Input()
+    public locale: string;
+
+    /**
+     * A custom formatter function, applied on the selected or passed in date.
+     *
+     * @example
+     * ```typescript
+     * private dayFormatter = new Intl.DateTimeFormat("en", { weekday: "long" });
+     * private monthFormatter = new Intl.DateTimeFormat("en", { month: "long" });
+     *
+     * public formatter(date: Date): string {
+     *  return `${this.dayFormatter.format(date)} - ${this.monthFormatter.format(date)} - ${date.getFullYear()}`;
+     * }
+     * ```
+     * ```html
+     * <igx-date-range-picker [formatter]="formatter"></igx-date-range-picker>
+     * ```
+     */
+    @Input()
+    public formatter: (val: DateRange) => string;
+
+    /**
+     * The default text of the calendar dialog `done` button.
+     *
+     * @remarks
+     * Default value is `Done`.
+     * The button will only show up in `dialog` mode.
+     *
+     * @example
+     * ```html
+     * <igx-date-range-picker [doneButtonText]="'完了'"></igx-date-range-picker>
```

Comment:
Fine as is, but we can simplify it
```suggestion
     * <igx-date-range-picker doneButtonText="完了"></igx-date-range-picker>
```